pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -278,6 +350,23 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         _startGradualWeightChange(startTime, endTime, _getNormalizedWeights(), endWeights, tokens);
     }
 
+    /**
+     * @dev Schedule a gradual swap fee change, from the current value to the given ending fee percentage,
+     * over startTime to endTime.
+     */
+    function updateSwapFeeGradually(
+        uint256 startTime,
+        uint256 endTime,
+        uint256 endSwapFeePercentage
+    ) external authenticate whenNotPaused nonReentrant {
+        _require(endSwapFeePercentage >= _getMinSwapFeePercentage(), Errors.MIN_SWAP_FEE_PERCENTAGE);
+        _require(endSwapFeePercentage <= _getMaxSwapFeePercentage(), Errors.MAX_SWAP_FEE_PERCENTAGE);
+
+        startTime = GradualValueChange.resolveStartTime(startTime, endTime);
+
+        _startGradualSwapFeeChange(startTime, endTime, super.getSwapFeePercentage(), endSwapFeePercentage);
```

Comment:
I've got some question marks on the usage of `super.getSwapFeePercentage()` here. It seems like we'll end up reading the start swap fee from `_miscData` rather than using the current swap fee. This'll work fine to begin with but we'll always revert back to the swap fee set in the last call to `_setSwapFeePercentage` or deployment.

Reply:
Fixed in [d103215](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1253/commits/d10321572a79df371f18b57382a88d3d5620894d)